src/binder/bind_node_visitor.cpp

```diff
@@ -440,6 +441,43 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::InsertStatement> node
           }
 
           ins_val->Accept(common::ManagedPointer(this).CastManagedPointerTo<SqlNodeVisitor>());
+
+          // if we have constant values of variable types being inserted into fixed size columns,
+          // we need to resolve that
+          if (ins_val->GetExpressionType() == parser::ExpressionType::VALUE_CONSTANT) {
+            auto const_val = ins_val.CastManagedPointerTo<parser::ConstantValueExpression>();
+            bool is_variable = const_val->GetReturnValueType() == type::TypeId::VARCHAR ||
+                               const_val->GetReturnValueType() == type::TypeId::VARBINARY;
+            if (is_variable && !const_val->IsNull() && ins_col.TypeModifier() != -1 &&
+                const_val->GetStringVal().GetLength() > static_cast<size_t>(ins_col.TypeModifier())) {
+              // postgres truncates off trailing spaces from the end of the varchar only to the max size of the col
+              // if it is still too large it throws an error
+              // https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/varchar.c
+              // issue #1380
+              size_t num_trailing_spaces = 0;
+              for (size_t idx = const_val->GetStringVal().GetLength() - 1;
+                   idx < const_val->GetStringVal().GetLength() && idx >= static_cast<size_t>(ins_col.TypeModifier());
```

Comment:
Why `idx < const_val->GetStringVal().GetLength()`? Can wraparound happen?

Reply:
Yeah if it wraps around then we have the whole thing as spaces? It can only happen if we have a size 0 column which idk if that can happen but will check.

Reply:
Size 0 should be impossible. I think I tried in Postgres `VARCHAR(0)` and it yells at you.

Reply:
done

Reply:
We also do not handle that error that postgres throws, I added that to the pr